src/appliance-onboarding-script/appliance_setup/avs/avsarconboarder/logger/_upload_logs.py

```
@@ -0,0 +1,63 @@
+import logging
+import pathlib
+import json
+
+from pkgs._exceptions import AzCommandError
+from pkgs._az_cli import az_cli
+from ..constants import Constant
+ 
+class UploadLogs:
+
+    def __init__(self, storage_account, config, managed_identity_id):
+        self.storage_account = storage_account
+        self.__config = config
+        self._storage_account_uri = Constant.STORAGE_ACCOUNT_URI
+        self.create_storage_blob_contributor_assignment(managed_identity_id)
+
+    def create_storage_blob_contributor_assignment(self, managed_identity_id):
+        storage_account_uri = self._storage_account_uri.format(self.__config["subscriptionId"], 
+                                                               self.__config["resourceGroup"],
+                                                               self.storage_account)
+
+        res, err = az_cli('role', 'assignment', 'create', 
+                        '--assignee', f'"{managed_identity_id}"',
+                        '--role', f'"Storage Blob Data Contributor"',
+                        '--scope', f'"{storage_account_uri}"')
+
+        if err:
```

> if err: [](http://example.com/codeflow?start=7&length=8)

Is this a fatal exception for Upload logs? If yes, should we raise an exception here?

---

We should always fail at the point of failure and not wait for the failure to propagate and fail at another point. This is because we get the most accurate error at this point.
Do verify the behavior if there is of the az cli in case the role assignment is pre-existing (This is in case of retries etc)

---

We do not get any error on retries(pre-existing role assignments). Logging it as error right now.

---

in that case should we raise an exception here?

---

yes let's keep an exception here.